bin/update-project-refs.js

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env node
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: loopback-next
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/**
+ * This is an internal script to update TypeScript project references based on
+ * lerna's local package dependencies.
+ *
+ * See https://www.typescriptlang.org/docs/handbook/project-references.html
+ */
+'use strict';
+
+const path = require('path');
+const fs = require('fs');
+const buildUtils = require('../packages/build/bin/utils');
+
+const Project = require('@lerna/project');
+const PackageGraph = require('@lerna/package-graph');
+
+const TSCONFIG = 'tsconfig.build.json';
+
+async function updateReferences() {
+  const project = new Project(process.cwd());
+  const packages = await project.getPackages();
+
+  const rootRefs = [];
+  const graph = new PackageGraph(packages);
+
+  for (const p of graph.values()) {
+    console.log('Package %s', p.pkg.name);
+    const pkgLocation = p.pkg.location;
+    const tsconfigFile = path.join(pkgLocation, TSCONFIG);
+    // Skip non-typescript packages
+    if (!fs.existsSync(tsconfigFile)) continue;
+    rootRefs.push({
+      path: path.join(
+        path.relative(project.rootPath, pkgLocation),
+        'tsconfig.build.json',
```

Comment:
Once we switch our monrepo to use project references, we should be able to get rid of `tsconfig.build.json` files and use regular per-package `tsconfig.json` file instead.

We are using `tsconfig.build.json` files to prevent VS Code from recognizing individual packages as individual TypeScript projects, and force it to treat our entire monorepo as a single TypeScript project. This is needed to make cross-package navigation and refactoring working.

With the new feature of project references, supported by VS Code too (now? in a near future?), there is no need to use a different configuration for VS Code and the compiler. We can rename per-package `tsconfig.build.json` files to regular `tsconfig.json` files.

We may be able to remove the repository-level `tsconfig.json` file too, but that requires changes in  the way how `tslint` is executed. I am not sure if running tslint on per-package basis is something we actually want to do.

Reply:
Ah, we cannot get rid of the top-level `tsconfig.json` file because this file must contain a list of references to all projects included in the monorepo.

Reply:
We can probably remove `tsconfig` files from git and generate them as part of the build.

Reply:
Sure, we can try that.

For now, I am fine with keeping the top-level tsconfig file in git. I think it may be easier for our users, as they don't have to run build before opening VS Code.